### PR TITLE
fix warning for homepage build

### PR DESCRIPTION
**Pull request by leckie-chn**
also 
* hide `sdk_reference.rst`
* fix source code deadlink w/ https global link
    * once issue https://github.com/rtfd/recommonmark/issues/137 fixed, will be able to reference relative path again. 

* fix deadlink for `README.md`

Also suggest that using links to homepage instead in `README.md`

### CODE REVIEW

File: docs/Trials.md [hunk 1]
```diff
@@ -30,38 +30,38 @@ Refer to [SearchSpaceSpec.md](./SearchSpaceSpec.md) to learn more about search s
...
 `metrics` could be any python object. If users use NNI built-in tuner/assessor, `metrics` can only have two formats: 1) a number e.g., float, int, 2) a dict object that has a key named `default` whose value is a number. This `metrics` is reported to [assessor](Builtin_Assessors.md). Usually, `metrics` could be periodically evaluated loss or accuracy.
 
 - Report performance of the configuration
 
-```json
+```python
 nni.report_final_result(metrics)
 ```
-`metrics` also could be any python object. If users use NNI built-in tuner/assessor, `metrics` follows the same format rule as that in `report_intermediate_result`, the number indicates the model's performance, for example, the model's accuracy, loss etc. This `metrics` is reported to [tuner](Builtin-Tuner.md).
+`metrics` also could be any python object. If users use NNI built-in tuner/assessor, `metrics` follows the same format rule as that in `report_intermediate_result`, the number indicates the model's performance, for example, the model's accuracy, loss etc. This `metrics` is reported to [tuner](tuners.md).
```
[the remark this answers is not in the document]
Comment (scarlett2018) [reply docs/Trials.md]: Another issue found in https://nni.readthedocs.io/en/latest/FAQ.html

refer is not found: https://github.com/Microsoft/nni/blob/master/docs/ExperimentConfig

[the remark this answers is not in the document]
Comment (scarlett2018) [reply docs/Trials.md]: https://nni.readthedocs.io/en/latest/KubeflowMode.html

Let's remove the email in the last para
> Any problems when using NNI in kubeflow mode, plesae create issues on NNI github repo, or send mail to nni@microsoft.com

[the remark this answers is not in the document]
Comment (scarlett2018) [reply docs/Trials.md]: https://nni.readthedocs.io/en/latest/KubeflowMode.html

The "here" in below para reached nowhere.
> We already build a docker image msranni/nni on Docker Hub. It contains NNI python packages, Node modules and javascript artifact files required to start experiment, and all of NNI dependencies. The docker file used to build this image can be found at here. You can either use this image directly in your config file, or build your own image based on it.

https://nni.readthedocs.io/en/deployment/Dockerfile.build.base

Got this:
 SORRY            /
         \                         /
          \    This page does     /
           ]   not exist yet.    [    ,'|
           ]                     [   /  |
           ]___               ___[ ,'   |
           ]  ]\             /[  [ |:   |
           ]  ] \           / [  [ |:   |
           ]  ]  ]         [  [  [ |:   |
           ]  ]  ]__     __[  [  [ |:   |
           ]  ]  ] ]\ _ /[ [  [  [ |:   |
           ]  ]  ] ] (#) [ [  [  [ :===='
           ]  ]  ]_].nHn.[_[  [  [
           ]  ]  ]  HHHHH. [  [  [
           ]  ] /   `HH("N  \ [  [
           ]__]/     HHH  "  \[__[
           ]         NNN         [
           ]         N/"         [
           ]         N H         [
          /          N            \
         /           q,            \
        /                           \

[the remark this answers is not in the document]
Comment (scarlett2018) [reply docs/Trials.md]: https://nni.readthedocs.io/en/latest/AdvancedNAS.html#weight-sharing-among-trials

"simple weight sharing example" goes no where.

> For details, please refer to this simple weight sharing example. We also provided a practice example for reading comprehension, based on previous ga_squad example.

[the remark this answers is not in the document]
Comment (leckie-chn) [reply docs/Trials.md]: actually it's `tuners.rst`, but recommonmark only recognize .md as hyperlink, and will convert it to `tuners.html` anyway.

[the remark this answers is not in the document]
Comment (leckie-chn) [reply docs/Trials.md]: @scarlett2018 for FAQ & email issue, please refer to https://github.com/Microsoft/nni/pull/680/commits/e439d42802b2a45d2a8de3c4e35faf93785e946c
and https://github.com/Microsoft/nni/pull/680/commits/1f5f04efa1a2a0cae2b3a8dc6e5c63fe1e83644f
Other deadlinks are already fixed in previous commits.

All those changes can take effect only after this PR is merged.

**Pull request merged**